dialoptions.go

```diff
@@ -413,6 +413,9 @@ func WithTimeout(d time.Duration) DialOption {
 // connections. If FailOnNonTempDialError() is set to true, and an error is
 // returned by f, gRPC checks the error's Temporary() method to decide if it
 // should try to reconnect to the network address.
+//
+// Note: Go overrides the OS defaults for TCP keepalive time and interval to 15s.
+// To retain OS defaults, use a net.Dialer with the KeepAlive field set to a negative value.
```

Comment:
```suggestion
// Note: Go overrides the OS defaults for TCP keepalive time and interval to 15s.
// To retain OS defaults, use a net.Dialer with the KeepAlive field set to a
// negative value.
```

Reply:
Let's say something like "As of Go 1.21, the standard library overrides ............"

Also, something that gives clarity into our default behavior would be nice, too.  E.g. "gRPC-Go's default dialer does this in order to restore the OS defaults."

Also:
```go
//
// For more information, please see [issue 48622] in the Go github repo.
//
// [issue 48622]: https://github.com/golang/go/issues/48622
```

Reply:
@dfawley actually the default TCP keepalive times were implemented since Go 1.13. The original issue in the Go github repo that discussed the implementation of default keepalive times is [here](https://github.com/golang/go/issues/23459). Here is the change I made for the documentation in grpc-go, do let me know if it is okay.

```
// Note: As of Go 1.13, the standard library overrides the OS defaults for
// TCP keepalive time and interval to 15s.
// To retain OS defaults, use a net.Dialer with the KeepAlive field set to a
// negative value.
//
// For more information, please see [issue 23459] in the Go github repo
//
// [issue 23459]: https://github.com/golang/go/issues/23459
```

Also, I am not really sure what you meant by grpc's default behavior. Based on my understanding, there has not been any restoration of OS defaults on grpc side until this PR. And the only restoration this PR can achieve is on the client side (see [this](https://github.com/grpc/grpc-go/pull/6672/files#diff-5334018444bcb34d15bcb3c8a3050cdf9c8a40fe41305c3a602ea4bfeb0cd178R181))